cs-minibot-platform-src/src/main/java/simulator/simbot/Dijkstras.java

```diff
@@ -0,0 +1,199 @@
+package simulator.simbot;
+
+import java.util.*;
+
+/**
+ * Created by jimmychen on 4/29/17.
+ */
+public class Dijkstras {
+
+    //public static int[][] path;
+    public static Node targetNode;
+    private static HashSet<Node> settledNodes = new HashSet<Node>();
+    private static HashSet<Node> unsettledNodes = new HashSet<Node>();
+    private static HashMap<Node, Integer> distances;
+    public static Map<Node, Node> predecessors = new HashMap<Node, Node>();
+
+    //Fills maze with 1's
+    public static int[][] initializeMaze(int n, int m) {
+        int[][] maze = new int[n][m];
+        for(int i = 0; i < n; i++) {
+            for(int j = 0; j < m; j++) {
+                maze[i][j] = 1;
+            }
+        }
+        //path = new int[n][m];
+        return maze;
+    }
+
+    //Takes the given maze and returns that maze wrapped in 1's
+    public static int[][] generateWrappedMaze(int[][] maze) {
+        int height = maze.length;
+        int width = maze[0].length;
+        int[][] wrappedMaze = initializeMaze(height+2,width+2);
+        for(int i = 1; i < height +1; i++) {
+            for(int j = 1; j < width + 1; j++) {
+                wrappedMaze[i][j] = maze[i-1][j-1];
+            }
+        }
+        return wrappedMaze;
+    }
+    //Makes the Nodes array but doesn't fill in neighbors
+    //Also sets cost to max int
+    public static Node[][] initializeNodeMatrix(int[][] maze) {
+        int height = maze.length;
+        int width = maze[0].length;
+        Node[][] res = new Node[height][width];
+        for(int i = 0; i < height; i++) {
+            for(int j = 0; j < width; j++) {
+                Node n = new Node(maze[i][j], false,i,j);
+                if(i == height-2 && j == width-2) {
+                    targetNode = n;
+                }
+                res[i][j] = n;
+                if(maze[i][j] == 0) {
+                    res[i][j].cost = Integer.MAX_VALUE;
+                }
+            }
+        }
+
+        res[height-2][width-2].isEnd = true;
+        res[1][1].cost = 1;
+        return res;
+    }
+    //Returns a HashSet of neighbors given a position and maze
+    public static HashSet<Node> generateNeighbors(int i, int j, Node[][] maze) {
+        HashSet<Node> res = new HashSet<Node>();
+        if(maze[i+1][j].value == 0) {
+            res.add(maze[i+1][j]);
+        }
+        if(maze[i][j+1].value == 0) {
+            res.add(maze[i][j+1]);
+        }
+        if(maze[i-1][j].value == 0) {
+            res.add(maze[i-1][j]);
+        }
+        if(maze[i][j-1].value == 0) {
+            res.add(maze[i][j-1]);
+        }
+        return res;
+    }
+
+    //Returns an array of Nodes with the neighbors field filled out
```

Comment:
it mutates the original array so it isn't necessary to return the array- people might think it is doing a copy.